ROADMAP.md

```diff
@@ -0,0 +1,57 @@
+# Roadmap
+
+The OpenAssetIO project is working towards an initial beta release of
+the core API.
+
+We are looking for early adopters who can help fine tune its final form,
+and validate key design choices, and help flesh out asset types and
+their properties.
+
+The project is functional, and moving forward we expect minimal churn in
+the API itself as we complete the implementation of language specific
+features.
+
+## Y23 Q3: v1.0.0b1
+
+This release will support resolution and publishing of entities in batch
+and interactive context, along with relationship discovery.
+
+**Hosts:** C++, Python
+**Managers:** Python
+
+- [x] Migration to the [Trait based API](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/doc/decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md).
+- [x] Migration to batch-centric API.
+- [-] Migration of core API methods to C+.
+- [ ] Sundry minor API Change issues.
+
+## Y24 Q1: v1.0.0
+
+This release will provide additional introspection methods and
+debugging functionality, along with language parity between Hosts and
+Managers.
+
+**Hosts:** C++, Python
+**Managers:** C++, Python
+
+- [ ] Auth related error codes.
```

Comment:
[consider] We also have a few places where we throw std exceptions (or let std exceptions propagate from library calls), when we should create our own exception types.  

Perhaps this is covered by "Sundry minor API Change issues" above (I'd argue that adding proper exceptions should go in the beta).

Reply:
I was lumping this more in the "to C++ migration" category, but we can call more of these things out if you like?

Reply:
I suppose calling out specifically about exception types is a bit low level for this doc, but perhaps something along the lines of "mapping out and concretely defining failure modes" perhaps?

Reply:
Yeah thats a good idea